app/controllers/trades_controller.rb

```diff
@@ -0,0 +1,77 @@
+class TradesController < ApplicationController
+  before_action :set_trade, only: %i[show edit update destroy]
+
+  # GET /trades or /trades.json
+  def index
+    @category = Category.find(params[:category_id])
+    @trades = @category.trades.order(created_at: :desc)
+  end
+
+  # GET /trades/1 or /trades/1.json
+  def show
+    @trade = Trade.find(params[:id])
+    @user = User.find(params[:user_id])
+  end
+
+  # GET /trades/new
+  def new
+    @trade = Trade.new
+    @categories = Category.all
+  end
+
+  # GET /trades/1/edit
+  def edit; end
+
+  # POST /trades or /trades.json
+  def create
+    @trade = Trade.new(trade_params)
+    # this associate the trade with the selected category
+    @trade.category_ids = params[:trade][:category_ids]
+    @trade.user = current_user
+    @categories = Category.all
+    respond_to do |format|
+      if @trade.save
+        format.html { redirect_to user_category_trades_path(@trade), notice: 'Trade was successfully created.' }
```

Comment:
- [ ] Great job wworking on this project. It really looks good. When I create a new trade and category, the notice: `Trade was successfully created` appears twice on the screen and is not good user experience. Kindly review why this happens and try to resolve this issue
![image](https://github.com/obed300/budget-app/assets/62849872/871b7bcd-31ab-4cdf-aa04-9708cc5a53ae)